### PR TITLE
I2338 if no explicit name given to a Problem, assign one automatically

### LINKED ISSUE #2338: Set Problem name automatically if user doesn't set it
### Summary of Issue

Some complex OpenMDAO scripts may create multiple Problem instances, and users are currently able to specify a name for each Problem instance, but it would be nice if we automatically provided a locally unique name for each Problem instance if the user didn't provide one.  This would allow OpenMDAO command line tools like the connection viewer and the n2 to target a specific Problem instance and the user wouldn't have to worry about it.  Problem instances would be named 'problem1', problem2', etc. based on their order of instantiation.

### Issue Type

- [ ] Bug
- [x] Enhancement

### PR DESCRIPTION
**Pull request by hschilling**
### Summary

Currently, if the user doesn't provide an explicit name for a Problem, the name is the empty string. If a model has sub problems, and the user wants to use one of the command line tools that take problem name as an argument, there is no way to indicate which Problem to apply the command to.

This PR assigns default values to Problems that are explicitly given a name.

### Related Issues

- Resolves #2338 

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: openmdao/core/problem.py [hunk 1]
```diff
@@ -138,8 +144,30 @@ def __init__(self, model=None, driver=None, comm=None, name=None, **options):
         """
         Initialize attributes.
         """
+        global _problem_counter
```
Comment (Kenneth-T-Moore) [inline openmdao/core/problem.py:147]: Do you also need to global `_problem_names ` since you are appending to it?

File: openmdao/core/problem.py [hunk 2]
```diff
@@ -138,8 +144,30 @@ def __init__(self, model=None, driver=None, comm=None, name=None, **options):
...
+
         self.cite = CITATION
-        self._name = name
+
+        # Code to give non-empty names to Problems so that they can be
+        # referenced from command line tools (e.g. check) that accept a Problem argument
+        _problem_counter += 1
```
Comment (Kenneth-T-Moore) [inline openmdao/core/problem.py:153]: Do we actually need `_problem_counter`, if the number of problems is always the length of `_problem_names`?

File: openmdao/core/problem.py [hunk 3]
```diff
@@ -138,8 +144,30 @@ def __init__(self, model=None, driver=None, comm=None, name=None, **options):
...
+        if name:  # if name hasn't been used yet, use it. Otherwise, error
+            if name not in _problem_names:
+                self._name = name
+            else:
+                raise ValueError(f"The problem name '{name}' already exists")
+        else:  # No name given: look for a name, of the form, 'problemN', that hasn't been used yet
+            _name = f"problem{_problem_counter}"
```
Comment (Kenneth-T-Moore) [inline openmdao/core/problem.py:160]: I think I understand the logic here now. If we are the second problem, but "problem2" is taken, we just start with "problem2.1", and keep working until we find one that isn't taken.

File: openmdao/core/tests/test_parallel_groups.py [hunk 4]
```diff
@@ -308,13 +308,11 @@ def test_setup_messages_bad_vec_type(self):
 
         # check that error is thrown if not using PETScVector
         if MPI:
-            msg = ("Problem: The `distributed_vector_class` argument must be a distributed vector "
+            msg = ("Problem .*: The `distributed_vector_class` argument must be a distributed vector "
```
Comment (Kenneth-T-Moore) [inline openmdao/core/tests/test_parallel_groups.py:311]: I am not sure where this ".*" comes from for the problem name, but it doesn't look so great. Seems like it should be "Problem Problem1", or maybe just "Problem".  There must be a bug in the print here that wasn't uncovered until you made the change.

**Review by Kenneth-T-Moore — CHANGES_REQUESTED** [review]

File: openmdao/core/problem.py (hunk 1 above)
Comment (hschilling) [reply openmdao/core/problem.py:147]: Good point. Fixed.

File: openmdao/core/problem.py (hunk 2 above)
Comment (hschilling) [reply openmdao/core/problem.py:153]: Also a great point. Will modify code to do that

File: openmdao/core/tests/test_parallel_groups.py (hunk 4 above)
Comment (Kenneth-T-Moore) [reply openmdao/core/tests/test_parallel_groups.py:311]: My bad, that's regex.

**Review by Kenneth-T-Moore — APPROVED** [review]

**Review by naylor-b — APPROVED** [review]

**Pull request merged**